Do not include <execinfo.h> when not available

On uClibc, <execinfo.h> is not necessarily installed, depending on the
configuration. Enlightenment properly makes <execinfo.h> optional, but
uses __GLIBC__ to check if it is available or not. Unfortunately,
uClibc defines __GLIBC__, so it is not the right way to decide if
<execinfo.h> is available.

Instead, Enlightenment already has a configure.ac check for
execinfo.h, so we simply use the result of this configure.ac check to
decide whether execinfo.h is to be included or not.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: b/src/bin/e.h
===================================================================
--- a/src/bin/e.h
+++ b/src/bin/e.h
@@ -92,7 +92,7 @@
 #  include <memcheck.h>
 # endif
 
-# ifdef __GLIBC__
+# ifdef HAVE_EXECINFO_H
 #  include <execinfo.h>
 # endif
 
